app/services/set_decline_by_default.rb

```diff
@@ -0,0 +1,33 @@
+class SetDeclineByDefault
+  def initialize(application_form:)
+    @application_form = application_form
+  end
+
+  def call
+    return false unless decided_choices.count == 3 && decided_choices.any?(&:offered_at)
```

Comment:
There can be fewer than three choices

Reply:
`decided_choices` covers both, offered and rejected.

But we need to fix the `decided_choices.count == 3` part, as you said there can be fewer that 3.

Reply:
Great :+1:. Is is right that we then go on to set `decline_by_default_at` for all the choices? Some of them may have been rejected, which makes DBD redundant

Reply:
we think this is not harmful, and it is a bit more futureproof, It would be possible, for example, to change an application from rejected to offered in case of a mistake from a provider.

Reply:
I think there's more to clarify here.

Our understanding is that provider offer/rejection is a binding decision and is not reversible. The UI does not support it.

If we set `decline_by_default_at` on *all* the applicationchoices, that puts the responsibility on the sweep job which comes round and does the declining (that is, the thing which will be implemented in the follow-up PR to this one) to know whether or not to do it. This means the logic for selecting applications to decline by default lives in two places: here and in the sweep. For this reason I would argue that it is better to only set the `dbd_at` and `dbd_days` on choices which will actually be declined.

Reply:
Ok, actually it should not take too much to implement it, thanks for clarify it.

Reply:
@duncanjbrown I've changed the logic so that the DBD fields are only set on offered application_choices, not rejected ones. The sweeper task will just need to look at ```:offered_at``` this way.